lib/oauth2.js

```diff
@@ -88,24 +88,35 @@ Strategy.prototype.authenticate = function(req, options) {
  * @api protected
  */
 Strategy.prototype.userProfile = function(accessToken, done) {
-  this._oauth2.get('https://www.googleapis.com/plus/v1/people/me', accessToken, function (err, body, res) {
+  this._oauth2.get('https://www.googleapis.com/oauth2/v3/userinfo', accessToken, function (err, body, res) {
     if (err) { return done(new InternalOAuthError('failed to fetch user profile', err)); }
 
     try {
       var json = JSON.parse(body);
 
       var profile = { provider: 'google' };
-      profile.id           = json.id;
-      profile.displayName  = json.displayName;
-      profile.name         = json.name;
+      profile.id           = json.id || json.sub;
+      profile.displayName  = json.name;
+      profile.name         = {
+        givenName: json.given_name,
+        familyName: json.family_name
+      };
```

Comment:
This (and the `displayName` change) make this an API breaking change. Not a big deal, but version bump should reflect this.

Reply:
https://github.com/mstade/passport-google-oauth2/pull/15/commits/b94250c2d3be003db149eec672ae54897cc79186